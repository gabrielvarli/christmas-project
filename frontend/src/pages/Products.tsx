// import { useOutletContext } from "react-router-dom";
// import ProductCard from "../components/ProductCard";
// import { ProductInfo } from "../models/productModels";
// import { getHeroPropsForPage } from "../components/Hero";
// import { HeroProps } from "../models/productModels";
// import Hero from "../components/Hero";
// import ErrorMessage from "../components/ErrorMessage";

// const Products = () => {
//   let products: ProductInfo[] = (useOutletContext() as ProductInfo[]) || [];
//   const heroInfo = getHeroPropsForPage("Products") as HeroProps;

//   return (
//     <div>
//       <Hero {...heroInfo} />
//       <div className="max-w-4xl text-center m-auto flex flex-wrap justify-center gap-8 my-12">
//         {products.length !== 0 ? (
//           products.map((product) => (
//             <ProductCard key={product.id} product={product} />
//           ))
//         ) : (
//           <ErrorMessage text="Couldn't fetch data..." />
//         )}
//       </div>
//     </div>
//   );
// };

// export default Products;

import { useOutletContext } from "react-router-dom";
import ProductCard from "../components/ProductCard";
import Hero from "../components/Hero";
import ErrorMessage from "../components/ErrorMessage";
import { ProductInfo, HeroProps } from "../models/productModels";
import { getHeroPropsForPage } from "../components/Hero";

const Products = () => {
  const products: ProductInfo[] = useOutletContext<ProductInfo[]>() || [];
  const heroInfo: HeroProps = getHeroPropsForPage("Products");

  return (
    <div>
      {/* Hero Section */}
      <Hero {...heroInfo} />

      {/* Products Section */}
      <div className="max-w-6xl mx-auto flex flex-wrap justify-center gap-8 py-12">
        {products.length > 0 ? (
          products.map((product) => (
            <ProductCard key={product.id} product={product} />
          ))
        ) : (
          <ErrorMessage text="No products available at the moment. Please check back later." />
        )}
      </div>
    </div>
  );
};

export default Products;
