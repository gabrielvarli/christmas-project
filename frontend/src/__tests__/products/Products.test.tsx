// import { render, screen } from "@testing-library/react";
// import "@testing-library/jest-dom";
// import Products from "../../pages/Products";
// import { createMemoryRouter, RouterProvider } from "react-router-dom";

// describe("Products page", () => {
//   beforeEach(() => {
//     const router = createMemoryRouter([{ path: "/", element: <Products /> }], {
//       initialEntries: ["/"],
//     });
//     render(<RouterProvider router={router} />);
//   });

//   it('should have a title heading stating "Vi erbjuder allt från Klassiska julgranar"', () => {
//     const heading: HTMLElement = screen.getByRole("heading", {
//       name: /Vi erbjuder allt från/i,
//     });
//     expect(heading).toBeInTheDocument();
//   });
//   it('should have a subtitle heading stating "Klassiska julgranar"', () => {
//     const heading: HTMLElement = screen.getByRole("heading", {
//       name: /Klassiska julgranar/i,
//     });
//     expect(heading).toBeInTheDocument();
//   });
// });

// testat med snapshot

import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Products from "../../pages/Products";
import { createMemoryRouter, RouterProvider } from "react-router-dom";

describe("Products page", () => {
  let container: HTMLElement;

  beforeEach(() => {
    const router = createMemoryRouter([{ path: "/", element: <Products /> }], {
      initialEntries: ["/"],
    });

    const renderResult = render(<RouterProvider router={router} />);
    container = renderResult.container; // Spara containern för snapshot-test
  });

  it('renders the title heading stating "Vi erbjuder allt från Klassiska julgranar"', () => {
    const heading = screen.getByRole("heading", {
      name: /Vi erbjuder allt från/i,
    });
    expect(heading).toBeInTheDocument();
  });

  it('renders the subtitle heading stating "Klassiska julgranar"', () => {
    const heading = screen.getByRole("heading", {
      name: /Klassiska julgranar/i,
    });
    expect(heading).toBeInTheDocument();
  });

  it("matches the snapshot", () => {
    expect(container).toMatchSnapshot();
  });
});
