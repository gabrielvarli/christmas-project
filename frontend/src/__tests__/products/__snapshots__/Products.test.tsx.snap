// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products page matches the snapshot 1`] = `
<div>
  <div>
    <section
      class="relative bg-[url('/images/framsidajulgran.png')] bg-cover bg-center bg-no-repeat max-h-[509px] flex items-center justify-center"
    >
      <div
        class="absolute inset-0 bg-gradient-to-b from-transparent via-black/30 to-black/50 z-0"
      />
      <div
        class="relative z-10 mx-auto max-w-screen-xl px-6 py-24 text-white"
      >
        <div
          class="text-center sm:text-left"
        >
          <h1
            class="text-4xl sm:text-5xl lg:text-6xl font-bold font-dancing leading-tight"
          >
            Vi erbjuder allt fr책n
            <strong
              class="block sm:text-6xl lg:text-7xl font-extrabold text-rose-400 mt-2"
            >
              Klassiska julgranar
            </strong>
          </h1>
          <p
            class="mt-4 text-lg sm:text-xl max-w-lg mx-auto sm:mx-0"
          >
            till moderna klassiska julgranar, ljusslingor och sm책, dekorativa detaljer
          </p>
        </div>
      </div>
    </section>
    <div
      class="max-w-6xl mx-auto flex flex-wrap justify-center gap-8 py-12"
    >
      <aside
        aria-live="polite"
        class="flex flex-col items-center text-center mt-20 p-4 bg-red-100 rounded-lg shadow-md"
        role="alert"
      >
        <h2
          class="font-dancing text-5xl text-red-600 mb-4"
        >
          Oh no!
        </h2>
        <p
          class="text-gray-800 mb-2"
        >
          N책got gick fel p책
           
          <span
            class="font-semibold"
          >
            Christmas Shopping!
          </span>
        </p>
        <p
          class="text-gray-700 italic"
        >
          No products available at the moment. Please check back later.
        </p>
        <img
          alt="Error illustration"
          class="mt-4 w-32 h-32 object-contain"
          src="/images/doftljus.png"
        />
      </aside>
    </div>
  </div>
</div>
`;
